Content.Server/Administration/Commands/BanMassCommand.cs

```diff
@@ -0,0 +1,117 @@
+        }
+
+        if (args.Length < 3)
+        {
+            shell.WriteLine(Loc.GetString("cmd-ban-invalid-arguments"));
+            shell.WriteLine(Help);
+            return;
+        }
+
+        reason = args[0];
+
+        if (!uint.TryParse(args[1], out minutes))
+        {
+            shell.WriteLine(Loc.GetString("cmd-ban-invalid-minutes", ("minutes", args[1])));
+            shell.WriteLine(Help);
+            return;
+        }
+
+        var player = shell.Player;
+        var targets = new List<string>();
+        var allTargets = argStr.Split(new[] { '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries);
+
+        foreach (var target in allTargets)
+        {
+            var trimmedTarget = target.Trim();
+            if (string.IsNullOrWhiteSpace(trimmedTarget))
+                continue;
+
+            var located = await _locator.LookupIdByNameOrIdAsync(trimmedTarget);
+
+            if (located == null)
+            {
+                shell.WriteError(Loc.GetString("cmd-ban-player", ("target", trimmedTarget)));
+                continue;
+            }
+
+            var targetUid = located.UserId;
+            var targetHWid = located.LastHWId;
+
+            var dbData = await _dbManager.GetAdminDataForAsync(targetUid);
+            if (dbData != null && dbData.AdminRank != null)
+            {
+                var targetPermissionsFlag = AdminFlagsHelper.NamesToFlags(dbData.AdminRank.Flags.Select(p => p.Flag));
+                if ((targetPermissionsFlag & AdminFlags.Permissions) == AdminFlags.Permissions) // Admins with Permission rights cannot be banned
+                    continue;
+            }
+
+            _bans.CreateServerBan(targetUid, trimmedTarget, player?.UserId, null, targetHWid, minutes, severity, reason);
+        }
+    }
+
+    public override CompletionResult GetCompletion(IConsoleShell shell, string[] args)
+    {
+        if (args.Length >= 3)
+        {
+            var options = _playerManager.Sessions.Select(c => c.Name).OrderBy(c => c).ToArray();
+            return CompletionResult.FromHintOptions(
+                options,
+                LocalizationManager.GetString("cmd-ban-hint"));
+        }
+
+        if (args.Length == 1)
+            return CompletionResult.FromHint(LocalizationManager.GetString("cmd-ban-hint-reason"));
+
+        if (args.Length == 2)
+        {
+            var durations = new CompletionOption[]
+            {
+                new("0", LocalizationManager.GetString("cmd-ban-hint-duration-1")),
+                new("1440", LocalizationManager.GetString("cmd-ban-hint-duration-2")),
+                new("4320", LocalizationManager.GetString("cmd-ban-hint-duration-3")),
+                new("10080", LocalizationManager.GetString("cmd-ban-hint-duration-4")),
+                new("20160", LocalizationManager.GetString("cmd-ban-hint-duration-5")),
+                new("43800", LocalizationManager.GetString("cmd-ban-hint-duration-6")),
```

Comment:
This is copy-pasted from `BanCommand.cs`. Ideally it should be defined in just one place and reused I think.

Reply:
I don't know exactly where this can be done

Reply:
I don't know where to do this

Reply:
Move the shared functionality between the two commands into some sort of helper function that can be called from both places. Do not copy paste things.